cellprofiler/modules/colortogray.py

```diff
@@ -528,14 +528,14 @@ def upgrade_settings(self,
                               setting_values[13:])
             variable_revision_number = 3
 
-        #
-        # Standardize the channel choices
-        #
-        setting_values = list(setting_values)
-        nchannels = int(setting_values[SLOT_CHANNEL_COUNT])
-        for i in range(nchannels):
-            idx = SLOT_FIXED_COUNT + SLOT_CHANNEL_CHOICE + i * SLOTS_PER_CHANNEL
-            channel_idx = self.get_channel_idx_from_choice(setting_values[idx])
-            setting_values[idx] = channel_idx+1
-
+        if variable_revision_number < 4:
```

Comment:
Great! I think this is exactly what was needed. I believe you'll also need to set `variable_revision_number` to 4 at the end of this block now (see line 529)